src/python.c

```diff
@@ -969,6 +969,49 @@ execute(ForeignScanState *node, ExplainState *es)
 		if(PyList_Size(p_pathkeys) > 0){
 			PyDict_SetItemString(kwargs, "sortkeys", p_pathkeys);
 		}
+        if (state->aggs)
+        {
+            PyObject *aggs = PyDict_New();
+            ListCell *lc_agg;
+            List *agg_list;
+
+            foreach(lc_agg, state->aggs)
+            {
+                PyObject    *agg,
+                            *function,
+                            *column;
+
+                agg = PyDict_New();
+
+                agg_list = (List *)lfirst(lc_agg);
+                function = PyUnicode_FromString(strVal(lsecond(agg_list)));
+                column = PyUnicode_FromString(strVal(lthird(agg_list)));
+
+                PyDict_SetItemString(agg, "function", function);
+                PyDict_SetItemString(agg, "column", column);
+                PyDict_SetItemString(aggs, strVal(linitial(agg_list)), agg);
+                Py_DECREF(agg);
+                Py_DECREF(function);
+                Py_DECREF(column);
+            }
+
+            PyDict_SetItemString(kwargs, "aggs", aggs);
+            Py_DECREF(aggs);
+        }
+        if (state->group_clauses)
+        {
+            PyObject *group_clauses = PyList_New(0);
+            ListCell *lc_groupc;
+
+            foreach(lc_groupc, state->group_clauses)
+            {
+                PyObject *column = PyUnicode_FromString(strVal(lfirst(lc_groupc)));
+                PyList_Append(group_clauses, column);
```

Comment:
I think (but not entirely sure, since https://docs.python.org/3/c-api/list.html#c.PyList_Append doesn't mention it -- some evidence in https://stackoverflow.com/questions/3512414/does-this-pylist-appendlist-py-buildvalue-leak) that PyList_Append increments the refcounter, so you need to DECREF the `column` here.

Reply:
Makes sense, added DECREF.